### ISSUE TITLE
bug: Manager refuses to launch

### ISSUE THREAD

**Issue by Hi-ImKyle** [ReVanced/revanced-manager#1469]
### Bug description

Originally had Revanced Manager installed but I saw that it had an update on the home screen of the app, so I clicked update and let it update itself. Relaunched the app after and it was just a black screen. So I cleared the cache, still a black screen. Reinstalled, still a black screen.

Seeing as I am running as Root. I have Scoop installed which attempts to catch any unhandled exceptions causing apps to crash, of which Scoop didn't notify me of anything.

Samsung S22 Ultra, Android 12, Magisk 26.1

[Manager Issue.webm](https://github.com/ReVanced/revanced-manager/assets/25551312/5141bd2b-32de-4ebf-ba78-11c8bf6f3ce8)

### Version of ReVanced Manager and version & name of application you tried to patch

Manager: 1.15.1

### Installation type

Root

### Device logs

```shell
n/a
```

### Patcher logs

```shell
n/a
```

### Acknowledgements

- [X] This request is not a duplicate of an existing issue.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] The issue is solely related to the ReVanced Manager

**Comment by oSumAtrIX** [comment]
Try reinstalling the app

**Comment by Hi-ImKyle** [comment]
> Try reinstalling the app

Yeah I said I already did, plus the video shows me doing exactly that..

**Comment by oSumAtrIX** [comment]
Can you open the page https://api.revanced.app

**Comment by Hi-ImKyle** [comment]
Opens fine on Desktop and Mobile just fine

**Comment by oSumAtrIX** [comment]
Use logcat and grep `" E "` by typing: `logcat | grep " E "`

**Comment by Hi-ImKyle** [comment]
```
11-05 21:54:11.072   936   981 E flutter : [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(error, Attempt to get length of null array, null, java.lang.NullPointerException: Attempt to get length of null array
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.createExceptionOrNull(Parcel.java:2444)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.createException(Parcel.java:2422)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.readException(Parcel.java:2405)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.readException(Parcel.java:2347)
11-05 21:54:11.072   936   981 E flutter :      at android.os.IDeviceIdentifiersPolicyService$Stub$Proxy.getSerialForPackage(IDeviceIdentifiersPolicyService.java:168)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Build.getSerial(Build.java:207)
11-05 21:54:11.072   936   981 E flutter :      at dev.fluttercommunity.plus.device_info.MethodCallHandlerImpl.onMethodCall(MethodCallHandlerImpl.kt:84)
11-05 21:54:11.072   936   981 E flutter :      at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:258)
11-05 21:54:11.072   936   981 E flutter :      at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
11-05 21:54:11.072   936   981 E flutter :      at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:322)
11-05 21:54:11.072   936   981 E flutter :      at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Handler.handleCallback(Handler.java:938)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Handler.dispatchMessage(Handler.java:99)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Looper.loopOnce(Looper.java:226)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Looper.loop(Looper.java:313)
11-05 21:54:11.072   936   981 E flutter :      at android.app.ActivityThread.main(ActivityThread.java:8751)
11-05 21:54:11.072   936   981 E flutter :      at java.lang.reflect.Method.invoke(Native Method)
11-05 21:54:11.072   936   981 E flutter :      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
11-05 21:54:11.072   936   981 E flutter :      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
11-05 21:54:11.072   936   981 E flutter : Caused by: android.os.RemoteException: Remote stack trace:
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.createExceptionOrNull(Parcel.java:2444)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.createException(Parcel.java:2422)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.readException(Parcel.java:2405)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.readException(Parcel.java:2347)
11-05 21:54:11.072   936   981 E flutter :      at com.android.internal.telephony.ITelephony$Stub$Proxy.getCarrierPrivilegeStatusForUid(ITelephony.java:11215)
11-05 21:54:11.072   936   981 E flutter :
11-05 21:54:11.072   936   981 E flutter : )
11-05 21:54:11.072   936   981 E flutter : #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:652)
11-05 21:54:11.072   936   981 E flutter : #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:310)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #2      MethodChannelDeviceInfo.deviceInfo (package:device_info_plus_platform_interface/method_channel/method_channel_device_info.dart:19)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #3      DeviceInfoPlugin.androidInfo (package:device_info_plus/device_info_plus.dart:54)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #4      ManagerAPI.getSdkVersion (package:revanced_manager/services/manager_api.dart:70)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #5      ManagerAPI.initialize (package:revanced_manager/services/manager_api.dart:64)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #6      main (package:revanced_manager/main.dart:20)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter :
```

**Comment by oSumAtrIX** [comment]
```
11-05 21:54:11.072   936   981 E flutter : Caused by: android.os.RemoteException: Remote stack trace:
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.createExceptionOrNull(Parcel.java:2444)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.createException(Parcel.java:2422)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.readException(Parcel.java:2405)
11-05 21:54:11.072   936   981 E flutter :      at android.os.Parcel.readException(Parcel.java:2347)
11-05 21:54:11.072   936   981 E flutter :      at com.android.internal.telephony.ITelephony$Stub$Proxy.getCarrierPrivilegeStatusForUid(ITelephony.java:11215)
11-05 21:54:11.072   936   981 E flutter :
11-05 21:54:11.072   936   981 E flutter : )
11-05 21:54:11.072   936   981 E flutter : #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:652)
11-05 21:54:11.072   936   981 E flutter : #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:310)
```

Looks like the issue originates somewhere outside of ReVanced Manager in some internal library.

**Comment by Hi-ImKyle** [comment]
Nothing has been updated except for Manager. No Apps, No Magisk Modules. Worked fine a few days ago. I can live with the CLI tool though if there's no other suggestions

**Comment by Ushie** [comment]
This could be related to the patch object migration code @oSumAtrIX

**Comment by oSumAtrIX** [comment]
It is not related to that because the migration code doesn't apply if you reinstall the app as well as because the exception stack shows this:
```
11-05 21:54:11.072   936   981 E flutter : #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:652)
11-05 21:54:11.072   936   981 E flutter : #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:310)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #2      MethodChannelDeviceInfo.deviceInfo (package:device_info_plus_platform_interface/method_channel/method_channel_device_info.dart:19)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #3      DeviceInfoPlugin.androidInfo (package:device_info_plus/device_info_plus.dart:54)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #4      ManagerAPI.getSdkVersion (package:revanced_manager/services/manager_api.dart:70)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #5      ManagerAPI.initialize (package:revanced_manager/services/manager_api.dart:64)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter : #6      main (package:revanced_manager/main.dart:20)
11-05 21:54:11.072   936   981 E flutter : <asynchronous suspension>
11-05 21:54:11.072   936   981 E flutter :
```

**Comment by Ushie** [comment]
@Hi-ImKyle What happens if you clear app data?

**Comment by Hi-ImKyle** [comment]
Messing around with settings. Literally found the problem when you pointed out it was related to something outside of the Manager app. Says `com.android.internal.telephony` so out of curiosity I went to check if I had my Phone app hidden from other apps and I did. So perhaps something was added in the last version or 2 that relies on such relation. Unhiding it makes Manager launch fine with no issues now. I'll leave this open just in case you want me to test anything though?

I cleared app data before, again shown in the video.
